File: Backend/Chess.Models/Entities/ChatMessage.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace Chess.Models.Entities
{
    public class ChatMessage : EntityBase
    {
        public UserBase User { get; set; }

        public string Text { get; set; }

        public LobbyConfig Lobby { get; set; }
    }
}

File: Frontend/src/app/models/chatMessage.ts
import { UserData } from "./userData";

export class ChatMessage {
    user: UserData;
    text: string;
    timeStamp: Date;

    constructor(_user: UserData, _text: string, _timeStamp:Date) {
        this.user = _user;
        this.text = _text;
        this.timeStamp = _timeStamp;
    }
}